_layouts/german.html

```diff
@@ -186,70 +181,123 @@
+                     document.getElementById(ids[i]).classList.add(class_item);
+                    }
+
+                 /* Case 4: we have mobile device that's neither ipad nor iphone
+                    and it has a large screen so probably a tablet*/
+                else if ('ontouchstart' in window && screen.width > 1024) {
+                     let class_item = ids[i] + "-tablet";
+                     document.getElementById(ids[i]).classList.add(class_item);
+                      /* we need to add more pdding if in portrait mode */
+                     if (IsPortrait && class_item == "program-grid-tablet") {
+                        $('.program-grid-tablet').css('padding-top', '35%');
+                        $('.program-grid-tablet').css('column-gap', '10%');
+                     }
+                   }
+
+                /* Case 5: Now this should be a desktop screen*/
+                else {
+                     let class_item = ids[i] + "-desktop";
+                     console.log('go here')
+                     document.getElementById(ids[i]).classList.add(class_item);
+                    }
+
+             }}
+    $(document).ready(function() {
+        // Call the function on page load
+        style_webpage();
+        // Call the function on orientation change
+        window.addEventListener("orientationchange", style_webpage);
+        // Call the function on orientation change
+        window.addEventListener("resize", style_webpage);
+        // Call the function on orientation change back
+        window.matchMedia("(orientation: portrait)").addListener(style_webpage);
+        // Call the function on orientation change back
+        window.matchMedia("(orientation: landscape)").addListener(style_webpage);
+    });
+
 </script>
 
-<!--
-<script>
-$(document).ready(function(){
-   var scroll_start = 0;
-   var item_list = $(document).getElementsByClassName('menu-4');
-   var sections = $(document).getElementsByTagName('section');
 
-   $(document).scroll(function() {
-    scroll_start = $(this).scrollTop();
-    var i = 0
-    for (var
+<script type="text/javascript">
+$(document).ready(function() {
+  /* Ipads have a strange pixel width so we check if we can identify them
+     otherwise they show up as mobile phone */
+  const IPA = /iPad|iPad4,1|iPad4,2|iPad5,1|iPad5,2|Macintosh/.test(navigator.userAgent);
+  let p = [];
+
+  if ('ontouchstart' in window && IPA) {
+    p = [
+      "{{ site.baseurl }}/images/boat-mobile.jpg",
+      "{{ site.baseurl }}/images/sunset-mobile.jpg",
+      "{{ site.baseurl }}/images/selfie-mobile.jpg",
+      "{{ site.baseurl }}/images/castle-mobile.jpg",
+    ];
+  } else if ('ontouchstart' in window && screen.width < 1024) {
+    p = [
+      "{{ site.baseurl }}/images/boat-mobile.jpg",
+      "{{ site.baseurl }}/images/sunset-mobile.jpg",
+      "{{ site.baseurl }}/images/selfie-mobile.jpg",
+      "{{ site.baseurl }}/images/castle-mobile.jpg",
+    ];
```

Comment:
You can merge this together, because you reference the same images:
``` Javascript
 if ('ontouchstart' in window && (IPA || screen.width < 1024)) {...}
```